examples/embedding/hello-embed.c

```diff
@@ -34,6 +34,7 @@
 #include "py/stackctrl.h"
 
 static char heap[16384];
+mp_state_ctx_t mp_state_ctx;
```

Comment:
This line shouldn't be needed.  I can build this example locally without this addition (but I do need the 2 changes above to Makefile.upylib).

Can you please test building again (from a clean directory) without this line?

Reply:
If I comment line 37 out and compile, I get this error:

```cc -std=c99 -I. -I../.. -DNO_QSTR   -c -o hello-embed.o hello-embed.c
cc -L.  hello-embed.o libmicropython.a   -o hello-embed
Undefined symbols for architecture x86_64:
  "_mp_state_ctx", referenced from:
      _gc_init in libmicropython.a(gc.o)
      _gc_lock in libmicropython.a(gc.o)
      _gc_unlock in libmicropython.a(gc.o)
      _gc_is_locked in libmicropython.a(gc.o)
      _gc_collect_start in libmicropython.a(gc.o)
      _gc_collect_root in libmicropython.a(gc.o)
      _gc_collect_end in libmicropython.a(gc.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [hello-embed] Error 1
```

Hmmm?

The definition is in py/mpstate.h as extern:

`282:extern mp_state_ctx_t mp_state_ctx;`

and it is defined in `mpstate.c` and should be there, and `objdump -t libmicropython.a` does show it **should** be defined:

```libmicropython.a(mpstate.o):    file format Mach-O 64-bit x86-64

SYMBOL TABLE:
0000000000000358         *COM*           00000008 _mp_state_ctx
```

I'm not familiar with the meaning of `COM` here. If I uncomment the line in hello-world.c an identical COM line is in the `hello-world.o` symbol table.

Any idea? Is this clang vs. gcc issue on symbol visibility?

Reply:
Strange... I'm not sure of the exact details but you can try these 2 things (independently):
1. add `-fno-common` to `CFLAGS`
2. make it `mp_state_ctx_t mp_state_ctx = {0};` in `py/mpstate.c`

Reply:
`-fno-common` worked. The symbol table is:

```
libmicropython.a(mpstate.o):    file format Mach-O 64-bit x86-64

SYMBOL TABLE:
0000000000000000 g     O __DATA,__common        _mp_state_ctx
```

`mp_state_ctx_t mp_state_ctx = {{0}};` worked. The symbol table for `mpstate.o` is:

```
libmicropython.a(mpstate.o):    file format Mach-O 64-bit x86-64

SYMBOL TABLE:
0000000000000000 g     O __DATA,__common        _mp_state_ctx
```
So either of those (I tested them in isolation) both fixed the symbol linking problem. Results from both are identical in the object file.

Reply:
Great, thanks for testing.  Let's go with the `-fno-common` solution.  Could you please update this PR with that fix?